### ISSUE TITLE
Suggestions

### ISSUE THREAD

**Issue by fschleifenbaum** [kagagnon/BEMBlade#1]
Hello there, great work!

I work a lot with BEM, I may recommend some changes for you:

- instead of having to define the block name AND the div container, maybe use the first @bem to render a div out of it and close it with @endbem
- add some tag before the block name automatically for better overview, such as "block_blockname" so you see in SCSS or LESS as well as in your rendered HTML it is a block

Again, great work - exactly what I was looking for! Thanks.

**Comment by kagagnon** [comment]
Hi,

Thank you for your comment, I really appreciate it.

For your first suggestion, I was thinking doing this for the initial release, but I couldn't wrap my head around on how to do it. There are so many factors why I made it this way: the tag name is not always a div, there can be other classes on the element (for instance, I use a different prefixed class for the JavaScript) and there can also be other attributes.

I could make the `@bem` function to accept 3 arguments: the tag name, the block name and an array of attributes. Does that seems like an interesting solution?

As for your second suggestion, I will probably not implement it since it would conflict with my own way to work. I already prefix my block element with what it represents. Example, I use `m_` if it's a module, `e_` if it's an element such as buttons and `l_` if it's a layout block.

```blade
@bem( 'm_login' )
@bem( 'e_button' )
@bem( 'l_section' )
```

Adding a prefix to those classes would hurt my workflow since the first letter also represents the file location of the block. That being said, if adding a prefix to your class names is a requirement, I may  add the feature as an option in the future!

**Comment by fschleifenbaum** [comment]
Hello,

thanks for considering my suggestions! For my first suggestion it would probably best to render a div by default and have the option to switch, like this:

```
@bem( 'cup') // Init Block "cup" as div
@bem( 'cup', 'ul') // Init Block "cup" as ul
```
For the second suggestion, maybe it is possible to set a default identifier in the config file? So if you don't set the value, it will be like before for manual usage.

Best regards

**Comment by kagagnon** [comment]
Those 2 features have been implemented in the [version 0.1.1](https://github.com/kagagnon/BEMBlade/tree/0.1.1).

Check it out and tell me if it fits your needs. I'm closing this issue for now
